pennylane/transforms/__init__.py

```diff
@@ -12,60 +12,89 @@
+transforms can be used to:
 
+- extract information from the quantum function (e.g., to extract unitary
+  matrices or draw the underlying circuit),
+
+- modify or compile the quantum function,
+
+- generate new quantum functions that compute quantum properties and metrics,
+  such as the gradient and the Fisher information matrix.
+
+This module provides a selection of device-independent, differentiable quantum
+gradient transforms. As such, not only is the output of a quantum transform
+differentiable, but quantum transforms *themselves* can be differentiated with
+respect to any floating point arguments.
+
+In addition, this module also includes an API for writing your own quantum
+transforms.
+
+.. note::
+
+    For gradient transforms, please see the :mod:`qml.gradients <pennylane.gradients>`
+    module.
 
 .. currentmodule:: pennylane
 
-Transforms
-----------
+Overview
+--------
 
-Transforms that act on QNodes
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Information transforms
+~~~~~~~~~~~~~~~~~~~~~~
 
 These transforms accept QNodes, and return new transformed functions
-that compute the desired quantity.
+that compute the desired quantity. These transforms do not result
+in quantum executions.
 
 .. autosummary::
     :toctree: api
 
-    ~transforms.classical_jacobian
-    ~batch_params
     ~draw
-    ~transforms.get_unitary_matrix
-    ~metric_tensor
     ~specs
-    ~transforms.mitigate_with_zne
+    ~transforms.get_unitary_matrix
+    ~transforms.classical_jacobian
 
-Transforms that act on quantum functions
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-These transforms accept quantum functions (Python functions
-containing quantum operations) that are used to construct QNodes.
+Multiple circuit transforms
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. autosummary::
-    :toctree: api
+These transforms accept QNodes, and return new QNodes
+that compute the desired quantity. The transformed QNode, when
+executed, may result in multiple quantum circuit evaluations
+under the hood.
+
+    ~batch_params
+    ~metric_tensor
+    ~transforms.mitigate_with_zne
+    ~transforms.hamiltonian_expand
+    ~transforms.measurement_grouping
 
-    ~adjoint
-    ~ctrl
-    ~apply_controlled_Q
-    ~quantum_monte_carlo
-    ~transforms.insert
 
-Transforms for circuit compilation
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Circuit modification and compilation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-This set of transforms accept quantum functions, and perform basic circuit compilation tasks.
+This set of transforms map a single quantum function to a new quantum
+function. This includes basic circuit compilation tasks.
 
 .. autosummary::
     :toctree: api
 
+    ~adjoint
+    ~ctrl
     ~compile
+    ~transforms.insert
     ~transforms.cancel_inverses
     ~transforms.commute_controlled
     ~transforms.merge_rotations
     ~transforms.single_qubit_fusion
     ~transforms.unitary_to_rot
+    ~apply_controlled_Q
```

Comment:
Not something to address here, but I think the first line of this docstring could be improved because right now it's not clear from the description what the subject area / purpose is unless you click on the link.

Reply:
(@trbromley)